src/Haskell/Ide/Engine/Transport/LspStdio.hs

```diff
@@ -730,8 +757,18 @@ reactor inp = do
 
 -- ---------------------------------------------------------------------
 
+-- | Queue a diagnostics request to be performed after a timeout. This prevents recompiling
+-- too often when there is a quick stream of changes.
 requestDiagnostics :: DiagnosticTrigger -> TrackingNumber -> J.Uri -> J.TextDocumentVersion -> R ()
-requestDiagnostics trigger tn file mVer = do
+requestDiagnostics dt tn uri mVer= do
```

Comment:
Given we now have `performDiagnostics`, `requestDiagnosticsNormal` and `requestDiagnostics`, it is getting confusing.

Perhaps we should do something like
- `requestDiagnostics` becomes `queueDiagnosticsRequest`
- `performDiagnostics` goes back to being `requestDiagnostics`

Reply:
Good idea